src/txHashBuilder.c

```diff
@@ -155,13 +155,10 @@ void txHashBuilder_enterInputs(tx_hash_builder_t* builder)
 	builder->state = TX_HASH_BUILDER_IN_INPUTS;
 }
 
-void txHashBuilder_addInput(
-        tx_hash_builder_t* builder,
-        const uint8_t* utxoHashBuffer, size_t utxoHashSize,
-        uint32_t utxoIndex
-)
+void txHashBuilder_addInput(tx_hash_builder_t *builder, tx_input_t input)
```

Comment:
We are generally passing arguments by pointers in such situations, I don't see a reason to copy the value here (it might even consume stack memory).